agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -2241,8 +2241,7 @@ bool slave_thread::handle_cmdu_ap_manager_message(Socket *sd,
             return false;
         }
 
-        channel_preference_tlv->radio_uid() =
-            network_utils::mac_from_string(config.radio_identifier);
+        channel_preference_tlv->radio_uid() = hostap_params.iface_mac;
```

Comment:
This will cause failures in the testbed unless we set the RUID in the UCC configuration file and the beerocks_agent.conf to be the interface MAC address, since the UCC will look for the RUID configured in the file in the channel preference TLV.

Bottom line - if we use the radio MAC as RUID, we need to do this everywhere. Otherwise things will break. Please do not merge this commit until we verify it in the testbed.

Reply:
Indeed, you also have to change the config file in the testbed.

I understood that this all-zeroes RUID was in fact an issue in the testbed, so I thought this would fix something. It is not strictly needed for the non-Intel agent test (I had an issue that I thought was caused by this, but it turned out to be something else), so I could also leave out this commit from the PR.

Reply:
Turns out that the correct RUID change is needed after all to make this PR work. So please adapt the certification setup config file so we can merge this PR.

Reply:
This will still fail in `dev_get_parameter` since in son_slave_thread `ACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION` we are still using the `config.radio_identifier` - so the VAPs list ruid will be wrong, which will be propagated to the agent ucc listener which eventually won't find the real RUID (the one from the ucc AllInitConfig.txt file) in the VAP list.

```
case beerocks_message::ACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION: {
        auto notification_in = beerocks_header->addClass<
            beerocks_message::cACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION>();
        if (notification_in == nullptr) {
            LOG(ERROR) << "addClass cACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION failed";
            return false;
        }

        LOG(INFO) << "received ACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION";

        auto notification_out = message_com::create_vs_message<
            beerocks_message::cACTION_CONTROL_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION>(cmdu_tx);
        if (notification_out == nullptr) {
            LOG(ERROR) << "Failed building message!";
            return false;
        }

        notification_out->params() = notification_in->params();
        LOG(TRACE) << "send ACTION_CONTROL_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION";
        send_cmdu_to_controller(cmdu_tx);

        auto notification_out2 = message_com::create_vs_message<
            beerocks_message::cACTION_BACKHAUL_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION>(cmdu_tx);
        if (notification_out2 == nullptr) {
            LOG(ERROR) << "Failed building message!";
            return false;
        }

        notification_out2->params() = notification_in->params();
        notification_out2->ruid()   = network_utils::mac_from_string(config.radio_identifier);
        LOG(TRACE) << "send ACTION_BACKHAUL_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION";
        message_com::send_cmdu(backhaul_manager_socket, cmdu_tx);

        break;
    }
```

Reply:
OK fair enough, I'll remove it completely in this PR then.